### DISCUSSION TITLE
Lazy Loading Components - Auto Import vs Lazy Import

### DISCUSSION THREAD

**Discussion by andorfermichael** [nuxt/nuxt#9342] [Questions (Nuxt 2)] [answered]
When using [auto-import ](https://github.com/nuxt/components#usage)does this automatically enable [lazy imports](https://github.com/nuxt/components#lazy-imports)?

I'm a bit confused because when reading the docs it sounds to me like if auto-import and lazy import were two different things. However, if I watch the network tab in the dev tools the components (chunks) are already imported (loaded) lazy asynchronously when the component is to be displayed. Due to that, if I rename components e.g. `my-component v-if="show"` to `lazy-my-component v-if="show"` nothing changes (improves further).

**Comment by pi0** [comment, marked as answer]
In compnents v2 / nuxt 2.15 and only for development, all components are lazily imported for speed (subject to change). Using `Lazy` prefix is correct usage for conditional lazy components and works with loader as well 😊 

More context: Nuxt also uses code splitting per page which means each route is lazy loaded. Even without Lazy prefix, compnents used in each page are lazy loaded alongside with page.

**Reply by andorfermichael** [reply]
so in which cases would I need lazy prefix then? Do I need them at all?

**Reply by pi0** [reply]
It is an optimization to be done with care (over-splitting js chunks has a cost of the wrapper and network overhead). A rule of thumb when to use the `Lazy` prefix:
- Component is conditionally rendered (with `v-if`), under the fold or is not an essential part of the page
- Component is heavy or has big dependencies (like a text editor or map component or lots of styles)
